### PR TITLE
Fix to plot_GCAM function: if statement that omits 'get_bbox_polys' f…

### PR DESCRIPTION
**Pull request by cledna**
…unction except when using Albers Equal-Area projections.

Otherwise it is unnecessary and distorts Antarctica in global projections.

This has been tested on the vignettes.

### CODE REVIEW

File: R/Map_Functions.R [hunk 1]
```diff
@@ -468,7 +468,14 @@ plot_GCAM <- function(mapdata, col = NULL, proj = robin, extent = EXTENT_WORLD,
 
     # Generate graticule (latitude/longitude lines) and clip map to extent specified.
     grat <- gen_grat()
-    mappolys <- get_bbox_polys(dataset = mapdata)
+    
+    # Only clip polygons if using an AEA projection; otherwise this can induce distortions
+    if(proj %in% c(na_aea, ch_aea)){
```
Comment (rplzzz) [inline R/Map_Functions.R:473]: This will not cover all Albers projections; it only gets the ones we have chosen to represent North America and China.  If this is truly an issue with the Albers projection, then we need to cover all variants of it.

**Review by rplzzz — CHANGES_REQUESTED** [review]
I am skeptical that this filtering is required only for the Albers projection.  If that's true, then _why_ does the Albers projection require it and not other, similar projections like, say, the Lambert?

For that matter, why is it unsafe to do this filtering with the global projections?  Without answers to these questions, we can't be sure that this fix will work in all cases, rather than just the ones you have tested.  We need to make sure we _really_ understand what is going on here.

On a related note, I have been looking at `get_bbox_polys`, and there are some puzzling things about it.  Although it takes an argument `bbox` for the bounding box, we never seem to use that argument, using the default `EXTENT_WORLD` bounding box instead.  Why is that.  Second, given that `EXTENT_WORLD` is currently `c(-180, 180, -90, 90)`, how is `get_bbox_polys` with that bounding box anything other than a no-op?  I contend that if filtering on `EXTENT_WORLD` produces any change at all in our map data, then there is something rotten in the map data.  Wouldn't you agree?

A side observation:  make sure that wherever we use `EXTENT_WORLD` as a default we are actually using the one we have defined.  In `coord_GCAM`, line 325--331 you use the global extent as a default, but you construct it yourself instead of using the defined version.  That's not good if we ever decide to change our global extent (e.g., to move it away from the poles a bit).

So, to summarize, we need to get to the bottom of whatever is going on here.  Why is an apparent no-op causing problems with one projection but required to fix a problem with another?  I don't think that "required for the Albers projection but messes up the others" is the right answer here.

File: R/Map_Functions.R (hunk 1 above)
Comment (cledna) [reply R/Map_Functions.R:473]: Ok. I agree that this is only a first-order fix that does not address what is really going on. To avoid problems with the global Antarctica maps while we solve the core issue, I propose an if-statement saying that if the projection is global, don't use the get_bbox_polys function. 
As for the get_bbox_polys function, I think what's happening is that we have data that is 180.00000 and a bounding box with 180 (those are the points I noticed that were getting clipped out). I think it may be a type issue. I don't think our map data is necessarily the problem, but I'll look into that more.

**Comment by rplzzz** [comment]
I've been looking at the the question of filtering, and it is quite a bit more complicated than the treatment that we've been giving it.  Basically, we have to figure out the boundaries of the desired view in the selected projection and filter to that.  The tricky thing is that the "boundaries of the view" are sometimes determined by the edges of the extent and sometimes determined by the limb of the earth.  My guess is there is a "right" way to do this, and we need to find out what that is, rather than trying to roll our own.  I will continue to investigate.
